discovery/discovery_test.go

```diff
@@ -248,36 +249,45 @@ func TestDiscoverEndpoints(t *testing.T) {
 
 	for i, tt := range tests {
 		httpDo = tt.do
+		httpDoInsecureTls = tt.do
 		var hostHeaders map[string]http.Header
 		if tt.authHeader != nil {
 			hostHeaders = map[string]http.Header{
 				strings.Split(tt.app.String(), "/")[0]: tt.authHeader,
 			}
 		}
-		de, _, err := DiscoverEndpoints(tt.app, hostHeaders, true)
-		if err != nil && !tt.expectDiscoverySuccess {
-			continue
-		}
-		if err != nil {
-			t.Fatalf("#%d DiscoverEndpoints failed: %v", i, err)
+		insecureList := []InsecureOption{
+			InsecureNone,
+			InsecureTls,
+			InsecureHttp,
+			InsecureTls | InsecureHttp,
 		}
+		for _, insecure := range insecureList {
+			de, _, err := DiscoverEndpoints(tt.app, hostHeaders, insecure)
+			if err != nil && !tt.expectDiscoverySuccess {
+				continue
+			}
+			if err != nil {
```

Comment:
Should we handle the case when `err == nil` and `tt.expectDiscoverySuccess == false`?